internal/services/streamanalytics/stream_analytics_output_cosmosdb_resource.go

```diff
@@ -0,0 +1,325 @@
+package streamanalytics
+
+import (
+	"context"
+	"fmt"
+	parse2 "github.com/hashicorp/terraform-provider-azurerm/internal/services/cosmos/parse"
+	validate2 "github.com/hashicorp/terraform-provider-azurerm/internal/services/cosmos/validate"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/streamanalytics/mgmt/2020-03-01-preview/streamanalytics"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/azure"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/streamanalytics/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/streamanalytics/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type OutputCosmosDBResource struct{}
+
+var _ sdk.ResourceWithCustomImporter = OutputCosmosDBResource{}
+
+type OutputCosmosDBResourceModel struct {
+	Name                  string `tfschema:"name"`
+	StreamAnalyticsJob    string `tfschema:"stream_analytics_job_name"`
+	ResourceGroup         string `tfschema:"resource_group_name"`
+	AccountKey            string `tfschema:"cosmosdb_account_key"`
+	Database              string `tfschema:"cosmosdb_sql_database_name"`
+	CollectionNamePattern string `tfschema:"collection_name_pattern"`
+	DocumentID            string `tfschema:"document_id"`
+	PartitionKey          string `tfschema:"partition_key"`
+}
+
+func (r OutputCosmosDBResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"stream_analytics_job_name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"resource_group_name": azure.SchemaResourceGroupName(),
+
+		"cosmosdb_account_key": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			Sensitive:    true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"cosmosdb_sql_database_name": {
```

Comment:
This should probably be renamed to `cosmosdb_sql_database_id` since that is all that we're accepting here.
```suggestion
		"cosmosdb_sql_database_id": {
```

Reply:
Will change.